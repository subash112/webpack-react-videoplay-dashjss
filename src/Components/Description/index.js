import React from 'react'

const Description = props => {
    return(
            <div>
                <ul>
                        <li>
                            <span>
                                <p>15</p>
                            </span>
                        </li>
                        <li>
                            <span>
                                <p>{props.duration} Minutes</p>
                            </span>
                        </li>
                        <li>
                            <span>
                                <p>{props.year}</p>
                            </span>
                        </li>
                </ul>
            <p>
                {props.plot}
            </p>

            </div>
    )
}

export default Description