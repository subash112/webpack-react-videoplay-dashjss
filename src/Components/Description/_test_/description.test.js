import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Description from '../index';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Description />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const { getByTestId } = render(
    <Description plot='This is an description section' />
  );
  expect(getByTestId('description').textContent).toBe(
    'This is an description section'
  );
});

it('matches snapshot', () => {
  const tree = TestRenderer.create(
    <Description plot='This is an description section' />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
