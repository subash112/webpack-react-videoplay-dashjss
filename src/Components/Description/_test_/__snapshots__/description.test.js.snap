// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="text-align-center p-1"
>
  <h2>
    Description
  </h2>
  <p
    className="p-1"
    data-testid="description"
  >
    This is an description section
  </p>
</div>
`;
